File: src/automatica.core/Automatica.WebNew/src/app/base/model/property-type.ts
import { BaseModel, JsonFieldInfo, JsonProperty, Model } from "./base-model"

@Model()
export class PropertyType extends BaseModel {

    @JsonProperty()
    Type: number;

    @JsonProperty()
    Name: string;

    @JsonProperty()
    Description: string;

    @JsonProperty()
    Meta: string;

    protected getJsonProperty(): Map<string, JsonFieldInfo> {
        return void 0;
    }

    public typeInfo(): string {
        return "PropertyType";
    }
}

File: src/automatica.core/Automatica.Core.EF/Models/PropertyType.cs
using System;
using System.Collections.Generic;
using Automatica.Core.Model;

namespace Automatica.Core.EF.Models
{
    public class PropertyType : TypedObject
    {

        public long Type { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Meta { get; set; }
        
    }
}
